plan/logical_plans.go

```diff
@@ -264,6 +264,11 @@ type DataSource struct {
 
 	LimitCount *int64
 
+	// inUpdate indicates the plan is in update or not.
+	// If it's true, its schema will contain writable columns,
+	// otherwise it will contain public columns.
+	inUpdate bool
```

Comment:
No, but I think keep this field is a good idea, so we won't forget the DataSource schema contains public columns or writable columns.